src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, [selectedTodo]);
+
+  const filteredTodos = useMemo(() => {
+    const lowerCaseQuery = query.toLowerCase();
+
+    return todos.filter(todo => {
+      const filteredByQuery = todo.title.toLowerCase().includes(lowerCaseQuery);
+
+      if (filter === 'active') {
+        return filteredByQuery && !todo.completed;
+      }
+
+      if (filter === 'completed') {
+        return filteredByQuery && todo.completed;
+      } else {
+        return filteredByQuery;
```

Comment:
Use a switch case construction instead of if conditions.
Also, create a enum for 'all', 'active', 'completed' and use it everywhere